npm installs fail if hardlinks not supported on file system

### What happened?

I'm running this on AKS using Azure Files for persistence, which doesn't support hard links. If I enable persistence on /data, I start getting errors when I try to run `npm install` (either through the manage palette page or running `npm i` manually).

Error log
```
2023-01-03T21:46:21.478Z npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production --engine-strict node-red-contrib-postgresql@0.11.3
2023-01-03T21:46:22.189Z [err] npm
2023-01-03T21:46:22.189Z [err]  WARN
2023-01-03T21:46:22.189Z [err]  config production Use `--omit=dev` instead.
2023-01-03T21:46:22.644Z [err] npm
2023-01-03T21:46:22.644Z [err]  ERR! code ENOTSUP
2023-01-03T21:46:22.644Z [err] npm
2023-01-03T21:46:22.644Z [err]  ERR!
2023-01-03T21:46:22.644Z [err]  syscall link
2023-01-03T21:46:22.644Z [err] npm ERR!
2023-01-03T21:46:22.644Z [err]  
2023-01-03T21:46:22.645Z [err] path /data/.npm/_cacache/tmp/78da15c8
2023-01-03T21:46:22.645Z [err] npm ERR! dest /data/.npm/_cacache/content-v2/sha512/71/6f/d7852a119e0ba4879feb7f584dc48ed54bd579c78f809b3af2188d89a3fc5fd14a421e419c3fa720905d8b7b5f2f42c76816907b0a160a50c78c3d57825b
2023-01-03T21:46:22.645Z [err] npm ERR! errno ENOTSUP
2023-01-03T21:46:22.647Z [err] npm ERR! Invalid response body while trying to fetch https://registry.npmjs.org/node-red-contrib-postgresql: ENOTSUP: operation not supported on socket, link '/data/.npm/_cacache/tmp/78da15c8' -> '/data/.npm/_cacache/content-v2/sha512/71/6f/d7852a119e0ba4879feb7f584dc48ed54bd579c78f809b3af2188d89a3fc5fd14a421e419c3fa720905d8b7b5f2f42c76816907b0a160a50c78c3d57825b'
2023-01-03T21:46:22.648Z [err] 
2023-01-03T21:46:22.648Z [err] npm ERR!
2023-01-03T21:46:22.648Z [err]  A complete log of this run can be found in:
2023-01-03T21:46:22.648Z [err] npm ERR!     /data/.npm/_logs/2023-01-03T21_46_22_092Z-debug-0.log
2023-01-03T21:46:22.674Z rc=1
```

### How can we reproduce this?

Here's my configuration

aks storage class (terraform config) and helm chart
```hcl
resource "kubernetes_storage_class" "nodered" {
  metadata {
    name = "nodered"
  }
  storage_provisioner = "file.csi.azure.com"
  reclaim_policy      = "Retain"
  parameters = {
    skuName = "Standard_LRS"
  }

  mount_options = ["dir_mode=0777", "file_mode=0777", "uid=10003", "gid=10003", "serverino", "mfsymlinks"]
}


resource "helm_release" "nodered" {
  chart      = "node-red"
  name       = "nodered"
  repository = "https://schwarzit.github.io/node-red-chart/"
  namespace  = var.namespace

  values = [
    replace(templatefile("${path.module}/values/nodered.values.yml", {
      domain             = "${var.nodered_subdomain}.${var.domain}"
      settings_config    = kubernetes_config_map.nodered_settings.metadata[0].name
      storage_class_name = kubernetes_storage_class.nodered.metadata[0].name
    }), "/^\\s*#.*\\n?/", "") # removes comment lines :)
  ]
}
```

`values.yaml`
```yaml
metrics:
  enabled: true
persistence:
  enabled: true
  storageClass: "${storage_class_name}"
ingress:
  enabled: true
  annotations: # {}
    kubernetes.io/ingress.class: traefik
    traefik.ingress.kubernetes.io/router.entrypoints: websecure,web
  hosts:
    - host: ${domain}
      paths:
        - path: /
          pathType: ImplementationSpecific
settings:
  name: nodered-settings-config
  configMapName: ${settings_config}
```

### Helm Chart version

```bash
0.21.3
```


### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

## Solution

I was able to solve it by:

1. allowing the user to write to `/usr/src/node-red`
```yaml
securityContext:
  readOnlyRootFilesystem: false
  runAsGroup: 1000
  runAsUser: 1000
```

2. writing this to `/usr/src/node-red/.npmrc`
```toml
cache="/usr/src/node-red/.cache"
```

But it would be great if your `npmrc:` config field accepted a free-text field (e.g. `npmrc.contents`) to insert values like this from the values.yaml file.

--- a/charts/node-red/Chart.yaml
+++ b/charts/node-red/Chart.yaml
@@ -9,7 +9,7 @@ icon: https://nodered.org/about/resources/media/node-red-icon-2.png
 
 type: application
 
-version: 0.21.3
+version: 0.22.0
 appVersion: 3.0.2
 
 keywords:
@@ -29,7 +29,7 @@ maintainers:
 annotations:
   artifacthub.io/containsSecurityUpdates: "false"
   artifacthub.io/changes: |
-    - remove replicas from values.yaml
+    - add option for custom content in npmrc file
   artifacthub.io/images: |
     - name: node-red
       image: docker.io/nodered/node-red:3.0.2

--- a/charts/node-red/README.md
+++ b/charts/node-red/README.md
@@ -1,6 +1,6 @@
 # node-red ⚙
 
-![Version: 0.21.3](https://img.shields.io/badge/Version-0.21.3-informational?style=for-the-badge) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=for-the-badge) ![AppVersion: 3.0.2](https://img.shields.io/badge/AppVersion-3.0.2-informational?style=for-the-badge)
+![Version: 0.22.0](https://img.shields.io/badge/Version-0.22.0-informational?style=for-the-badge) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=for-the-badge) ![AppVersion: 3.0.2](https://img.shields.io/badge/AppVersion-3.0.2-informational?style=for-the-badge)
 
 [![Artifact Hub](https://img.shields.io/endpoint?url=https://artifacthub.io/badge/repository/node-red&style=for-the-badge)](https://artifacthub.io/packages/search?repo=node-red)
 [![SIT](https://img.shields.io/badge/SIT-awesome-blueviolet.svg?style=for-the-badge)](https://jobs.schwarz)
@@ -16,7 +16,7 @@ A Helm chart for Node-Red, a low-code programming for event-driven applications
 To install the chart using the OCI artifact, run:
 
 ```bash
-helm install node-red oci://ghcr.io/schwarzit/charts/node-red --version 0.21.3
+helm install node-red oci://ghcr.io/schwarzit/charts/node-red --version 0.22.0
 ```
 
 ## Usage
@@ -32,7 +32,7 @@ helm repo update
 To install the chart with the release name node-red run:
 
 ```bash
-helm install node-red node-red/node-red --version 0.21.3
+helm install node-red node-red/node-red --version 0.22.0
 ```
 
 After a few seconds, node-red should be running.
@@ -93,6 +93,7 @@ The command removes all the Kubernetes components associated with the chart and 
 | metrics.serviceMonitor.selector | object | `{}` | Prometheus ServiceMonitor selector |
 | nameOverride | string | `""` | Provide a name in place of node-red |
 | nodeSelector | object | `{}` | Node selector |
+| npmrc.content | string | `"# Custom npmrc config\n"` | Configuration to add custom npmrc config |
 | npmrc.enabled | bool | `false` | Enable custom npmrc config |
 | npmrc.registry | string | `"https://registry.npmjs.org"` | Configuration to use any compatible registry |
 | persistence.accessMode | string | `"ReadWriteOnce"` | Persistence access mode |

--- a/charts/node-red/templates/npmrc-cm.yaml
+++ b/charts/node-red/templates/npmrc-cm.yaml
@@ -5,5 +5,10 @@ metadata:
   name: {{ include "node-red.configMapName" . }}
 data:
   npmrc: |
+{{- if .Values.npmrc.registry }}
     registry={{ .Values.npmrc.registry }}
+{{- end }}
+{{- if .Values.npmrc.content }}
+    {{- .Values.npmrc.content | nindent 4 }}
+{{- end }}
 {{- end -}}

--- a/charts/node-red/values.yaml
+++ b/charts/node-red/values.yaml
@@ -28,6 +28,9 @@ npmrc:
   enabled: false
   # -- Configuration to use any compatible registry
   registry: "https://registry.npmjs.org"
+  # -- Configuration to add custom npmrc config
+  content: |
+    # Custom npmrc config
 
 # -- node-red env, see more environment variables in the [node-red documentation](https://nodered.org/docs/getting-started/docker)
 env: []

